### PR TITLE
Add PDE experiment

### PR DESCRIPTION
**Pull request by yb6599**

### CODE REVIEW

**Comment by Jacob-Stevens-Haas** [comment]
To recap and expand on what I just said in person:
> 1. Remove debug.py and any extra commented out code
> 2. Get gen_experiments.pdes.run_pde() working with an example posted here.

At that point I'll be ready to help with the merge conflicts

File: src/gen_experiments/__init__.py [hunk 1]
```diff
@@ -466,6 +466,7 @@ def _convert_abs_rel_noise(grid_vals: list, grid_params: list, recent_results: d
         diff_params["kalman"],
         ["diff_params.alpha"],
         [(None,)],
+
```
Comment (yb6599) [inline src/gen_experiments/__init__.py:469]: Remove this line

File: src/gen_experiments/__init__.py [hunk 2]
```diff
@@ -598,3 +599,9 @@ def identity(x):
         [this_module],
     ),
 }
+skinny_specs = {
+    "exp3": (
+        ("sim_params.noise_abs", "diff_params.meas_var"),
+        (lambda x: x, lambda x: x)
+    )
+}
```
Comment (yb6599) [inline src/gen_experiments/__init__.py:607]: EOF new line

File: src/gen_experiments/pdes.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+            metrics, {
+                "t_train": t_train,
+                "x_train": x_train,
+                "x_test": x_test,
+                "x_dot_test": x_dot_test,
+                "x_train_true": x_train_true,
+                "model": model,
+            }
+        )
+    return metrics
```
Comment (yb6599) [inline src/gen_experiments/pdes.py:235]: EOF new line

File: src/gen_experiments/utils.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 from collections import defaultdict
 from dataclasses import dataclass, field
 from itertools import chain
-from types import ModuleType
+from types import ModuleTypebranch
```
Comment (yb6599) [inline src/gen_experiments/utils.py:4]: Revert

File: src/gen_experiments/utils.py [hunk 5]
```diff
@@ -150,7 +150,7 @@ def gen_pde_data(
     t_end: int = 100,
 ):
     """Generate PDE measurement data for training
-
+
```
Comment (yb6599) [inline src/gen_experiments/utils.py:153]: Revert

**Comment by yb6599** [comment]
I've created a new branch with all the unimplemented PDEs and this branch only contains "diffuse1D" PDE and it should run using
`gen_experiments.pdes.run_pde(42,
        group= "diffuse1D",
        diff_params=gen_experiments.diff_params["test_axis"],
        opt_params=gen_experiments.opt_params["test"],
        feat_params=gen_experiments.feat_params["pde"],
)`

**Comment by Jacob-Stevens-Haas** [comment]
You asked this earlier, but you need to add `from gen_experiments import pdes` to `__init__.py`.  Otherwise I get:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'gen_experiments' has no attribute 'pdes'. Did you mean: 'odes'?
```

File: src/gen_experiments/__init__.py (hunk 2 above)
Comment (Jacob-Stevens-Haas) [reply src/gen_experiments/__init__.py]: Still relevant

File: src/gen_experiments/pdes.py (hunk 3 above)
Comment (Jacob-Stevens-Haas) [reply src/gen_experiments/pdes.py]: still relevant

**Review by Jacob-Stevens-Haas — CHANGES_REQUESTED** [review]
Hey Yash, 
See the two old changes you noted in the review a few weeks ago about EOF.  Those, and getting this working with 
```
python -m gen_experiments pde ...
```
Is all that's needed for the PR

File: src/gen_experiments/__init__.py (hunk 2 above)
Comment (yb6599) [reply src/gen_experiments/__init__.py]: That's weird, my local files have an EOF line and I've commited and pushed those changes, but those changes are not showing up here in Files changed.

**Comment by yb6599** [comment]
Hey Jake,
I've got 
`python -m gen_experiments ...`
working for ODEs and PDEs.

**Comment by Jacob-Stevens-Haas** [comment]
Great!  Write full `python -m gen_experiments ...` command here so that I can attempt it on my machine.

File: src/gen_experiments/__init__.py (hunk 2 above)
Comment (Jacob-Stevens-Haas) [reply src/gen_experiments/__init__.py]: check `git config core.autocrlf`.  It should be set to `auto`.  If that's the case and you're still showing this issue, I'll go ahead and merge this anyways.

Comment (yb6599) [reply src/gen_experiments/__init__.py]: `git config core.autocrlf` returns `true`

**Comment by yb6599** [comment]
For ODEs,
`python -m gen_experiments lorenz --seed 23 --param sim_params=test --param diff_params=test --param opt_params=test --param feat_params=test`
For PDEs,
`python -m gen_experiments diffuse1D --seed 54 --param diff_params=test_axis --param opt_params=test --param feat_params=pde`

**Comment by Jacob-Stevens-Haas** [comment]
Ok, it looks like the pdes _almost_ works.  When I run it, I get a lot more output than running the lorenz example.  This is probably a bug in mitosis, but I want to make sure.

File: src/gen_experiments/pdes.py (hunk 3 above)
Comment (Jacob-Stevens-Haas) [reply src/gen_experiments/pdes.py]: NVM I inspected the hexdump - this is just how github renders files.

File: src/gen_experiments/__init__.py (hunk 2 above)
Comment (Jacob-Stevens-Haas) [reply src/gen_experiments/__init__.py]: Ok that's apparently how it's supposed to be in git for windows. :+1:

File: src/gen_experiments/pdes.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+def run(
+    seed: float,
+    /,
+    group: str,
+    diff_params: dict,
+    feat_params: dict,
+    opt_params: dict,
+    display: bool = True,
+    return_all: bool = True,
```
Comment (Jacob-Stevens-Haas) [inline src/gen_experiments/pdes.py:47]: default to `False` please, this is used to turn on a much larger return dictionary when used inside another experiment that builds composite plots (e.g. gridsearch).  This is what's causing the much larger output.

**Review by Jacob-Stevens-Haas — CHANGES_REQUESTED** [review]
LGTM just a minor tweak.  Forget about the line ending stuff, I verified it works with `hexdump` (inspecting file binary contents)

**Pull request merged**